### DISCUSSION TITLE
Queue binding sometimes breaks intransparently

### DISCUSSION THREAD

**Discussion by ghost** [rabbitmq/rabbitmq-server#3540] [Questions]
Running a three node cluster with rabbitmq version 3.8.18 deployed with the kubernetes operator I use the following topology configuration:
 
-  A durable topic exchange
- A transient classic queue
- A binding of the queue to the exchange with routing key '*.*'

This generally works but from time to time the binding just seems to break. Though the exchanges continues receiving messages, no messages are passed to the queue so it just remains empty. The issue sometimes occurs after one ore more of the rabbitmq nodes are restarted and sometimes just during normal operation.

I couldn't find any error message in the rabbitmq logs or any other indication that something is wrong. Also I coulndt find a way to reproduce the issue, it just happens from time to time (several times a year so not too often). The only way to correct the issue I found is to manually delete the binding on the queue and create it again with the same parameters.

Did anyone here come across a similiar issue or could help me how to debug this further should it occur again?

**Comment by vikinghawk** [comment]
Have definitely seen this before on clusters with higher numbers of these transient queues. Its due to a timing issue on the server between cleanup of the transient queue and the client coming back and recreating it.

We worked around this issue by having client connection recovery rename the queue:
https://github.com/rabbitmq/rabbitmq-java-client/pull/693
